addons/mrp/migrations/10.0.2.0/post-migration.py

```diff
@@ -278,37 +284,112 @@ def populate_stock_move_workorder_id(cr):
+            'field': quantity_done_field,
+            'field_type': 'float',
+        })
 
-    for production in productions.filtered(
-            lambda r: r.state not in ['done', 'cancel']):
-        diff = production.product_qty - production.qty_produced
-        cr.execute(
-            """
+    # compute quantity_done for product lots
+
+    openupgrade.logged_query(cr, """
+        UPDATE stock_move sm
+        SET %(field)s = lot_qty.quantity_done
+        FROM (
+        SELECT sum(COALESCE(sml.quantity_done, 0)) as quantity_done,
+            sml.move_id
+        FROM stock_move sm
+        INNER JOIN stock_move_lots sml on sml.move_id = sm.id
+        INNER JOIN stock_production_lot spl on spl.id = sml.lot_id
+        INNER JOIN product_product pp on pp.id = sm.product_id
+        INNER JOIN product_template pt on pt.id = pp.product_tmpl_id
+        WHERE (pt.tracking != 'none' OR  sml.lot_id IS NOT NULL)
+            AND sml.done_wo = True
+        GROUP BY sml.move_id) AS lot_qty
+        WHERE sm.id = lot_qty.move_id AND %(field)s IS NULL
+        ;
+        """ % {
+        'field': quantity_done_field,
+    })
+    # compute quantity_done for not product lots
+
+    openupgrade.logged_query(cr, """
+        UPDATE stock_move sm
+        SET %(field)s = quantity_done_store
+        FROM
+        product_product pp
+        INNER JOIN product_template pt ON pt.id = pp.product_tmpl_id
+        WHERE pp.id = sm.product_id
+        AND (pt.tracking = 'none' or pt.tracking IS NULL)
+        AND %(field)s IS NULL;
+        """ % {
+        'field': quantity_done_field,
+    })
+
+    qty_produced_field = 'qty_produced'
+    if not openupgrade.column_exists(cr, 'mrp_production', 'qty_produced'):
+        qty_produced_field = openupgrade.get_legacy_name('qty_produced')
+        openupgrade.logged_query(cr, """
+            ALTER TABLE %(table_name)s
+            ADD COLUMN %(field)s %(field_type)s;
+            """ % {
+            'table_name': 'mrp_production',
+            'field': qty_produced_field,
+            'field_type': 'float',
+        })
+
+    # compute qty_produced for mrp_production
+
+    openupgrade.logged_query(cr, """
+        UPDATE mrp_production mp
+        SET %(qty_produced_field)s = qty_done_tb.quantity_done
+        FROM (
+        SELECT sum(COALESCE(sm.%(quantity_done_field)s, 0)) as quantity_done,
+            sm.production_id FROM
+        stock_move sm
+        INNER JOIN mrp_production mp on sm.production_id = mp.id
+        WHERE sm.scrapped = False AND sm.state != 'cancel'
+        GROUP BY sm.production_id)
+        as qty_done_tb
+        WHERE qty_done_tb.production_id = mp.id
+        AND %(qty_produced_field)s IS NULL;
+        """ % {
+        'quantity_done_field': quantity_done_field,
+        'qty_produced_field': qty_produced_field,
+    })
+
+    mp_obj = env['mrp.production']
+    domain = [('state', 'not in', ['done', 'cancel'])]
+    offset = 0
+    while len(mp_obj.search(
```

Comment:
Chunking here is slower than just running the query once across all open production orders.